src/mac/frame/mod.rs

```diff
@@ -50,263 +49,183 @@ pub struct Frame<'p> {
+/// // invalid, for the sake of convenience.
+/// let bytes = [
+///     0x01u8, 0x98,             // frame control
+///     0x00,                   // sequence number
+///     0x12, 0x34, 0x56, 0x78, // PAN identifier and address of destination
+///     0x12, 0x34, 0x9a, 0xbc, // PAN identifier and address of source
+///     0xde, 0xf0,             // payload
+///     0x12, 0x34,             // payload
+/// ];
+///
+/// let frame: Frame = bytes.read_with(&mut 0, FooterMode::Explicit).unwrap();
+/// let header = frame.header;
+///
+/// assert_eq!(frame.header.seq,       0x00);
+/// assert_eq!(header.frame_type,      FrameType::Data);
+/// assert_eq!(header.security,        Security::None);
+/// assert_eq!(header.frame_pending,   false);
+/// assert_eq!(header.ack_request,     false);
+/// assert_eq!(header.pan_id_compress, false);
+///
+/// assert_eq!(
+///     frame.header.destination,
+///     Some(Address::Short(PanId(0x3412), ShortAddress(0x7856)))
+/// );
+/// assert_eq!(
+///     frame.header.source,
+///     Some(Address::Short(PanId(0x3412), ShortAddress(0xbc9a)))
+/// );
+///
+/// assert_eq!(frame.payload, &[0xde, 0xf0]);
+///
+/// assert_eq!(frame.footer, [0x12, 0x34]);
+/// #
+/// # Ok(())
+/// # }
+/// ```
+/// Encodes the frame into a buffer
+///
+/// # Example
+///
+/// ## allocation allowed
+/// ``` rust
+/// use ieee802154::mac::{
+///   Frame,
+///   FrameContent,
+///   FooterMode,
+///   Address,
+///   ShortAddress,
+///   FrameType,
+///   FrameVersion,
+///   Header,
+///   PanId,
+///   Security,
+/// };
+/// use byte::BytesExt;
+///
+/// let frame = Frame {
+///     header: Header {
+///         frame_type:      FrameType::Data,
+///         security:        Security::None,
+///         frame_pending:   false,
+///         ack_request:     false,
+///         pan_id_compress: false,
+///         version:         FrameVersion::Ieee802154_2006,
+///
+///         seq:             0x00,
+///         destination: Some(Address::Short(PanId(0x1234), ShortAddress(0x5678))),
+///         source:      Some(Address::Short(PanId(0x1234), ShortAddress(0x9abc))),
+///     },
+///     content: FrameContent::Data,
+///     payload: &[0xde, 0xf0],
+///     footer:  [0x12, 0x34]
+/// };
+///
+/// // Work also with `let mut bytes = Vec::new()`;
+/// let mut bytes = [0u8; 32];
+/// let mut len = 0usize;
+///
+/// bytes.write_with(&mut len, frame, FooterMode::Explicit).unwrap();
+///
+/// let expected_bytes = [
+///     0x01, 0x98,             // frame control
+///     0x00,                   // sequence number
+///     0x34, 0x12, 0x78, 0x56, // PAN identifier and address of destination
+///     0x34, 0x12, 0xbc, 0x9a, // PAN identifier and address of source
+///     0xde, 0xf0,             // payload
+///     0x12, 0x34              // footer
+/// ];
+/// assert_eq!(bytes[..len], expected_bytes);
+/// ```
+///
+/// Tells [`Frame::encode`] whether to read/write the footer
```

Comment:
Since [`Frame::encode`] no longer exists, this no longer fits. Maybe this instead:
```suggestion
/// Controls whether the footer is read/written with the frame
```

Reply:
Done